src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -5,91 +5,161 @@
-        [CLSCompliant(false)]
-        public void SetUInt64(ulong value) { }
+        public override string ToString() => _value; 
+        
+        public byte GetByte() => byte.Parse(_value);
+        
+        public short GetInt16() => short.Parse(_value);
+        
+        public int GetInt32() => int.Parse(_value);
+        
+        public long GetInt64() => long.Parse(_value);
+        
+        public float GetSingle() => float.Parse(_value);
+        
+        public double GetDouble() => double.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public sbyte GetSByte() => sbyte.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public ushort GetUInt16() => ushort.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public uint GetUInt32() => uint.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public ulong GetUInt64() => ulong.Parse(_value);
+
+        public bool TryGetByte(out byte value) => byte.TryParse(_value, out value);
+
+        public bool TryGetInt16(out short value) => short.TryParse(_value, out value);
+
+        public bool TryGetInt32(out int value) => int.TryParse(_value, out value);
+
+        public bool TryGetInt64(out long value) => long.TryParse(_value, out value);
+
+        public bool TryGetSingle(out float value) => float.TryParse(_value, out value);
+
+        public bool TryGetDouble(out double value) => double.TryParse(_value, out value);
 
-        public static implicit operator JsonNumber(byte value) { throw null; }
-        public static implicit operator JsonNumber(int value) { throw null; }
-        public static implicit operator JsonNumber(short value) { throw null; }
-        public static implicit operator JsonNumber(long value) { throw null; }
-        public static implicit operator JsonNumber(float value) { throw null; }
-        public static implicit operator JsonNumber(double value) { throw null; }
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(sbyte value) { throw null; }
+        public bool TryGetSByte(out sbyte value) => sbyte.TryParse(_value, out value);
+
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(ushort value) { throw null; }
+        public bool TryGetUInt16(out ushort value) => ushort.TryParse(_value, out value);
+
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(uint value) { throw null; }
+        public bool TryGetUInt32(out uint value) => uint.TryParse(_value, out value);
+
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(ulong value) { throw null; }
+        public bool TryGetUInt64(out ulong value) => ulong.TryParse(_value, out value);
+
+
+        public void SetFormattedValue(string value)
+        {
+            if (value == null)
+                throw new NullReferenceException("Expected number, but instead got null.");
+            if (value == "")
```

Comment:
nit: Checking length will likely be much faster (and avoid a call to `.Equals`). [sharplab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUXAD2bANuQLJ4AUWPAAZyANwicAlPUbN5WAOzjJ5ALxryAIi0BuOUwC+BluXZdeOQSJXTZReQuUTOAOgAycAHYBzGAAt1TWF9B2ZjIkMgA===) 
```suggestion
            if (value.Length == 0)
```

Reply:
Or combine both with `string.IsNullOrEmpty(value)`

Reply:
I split those cases to throw different exceptions.